nix: cargo-nextest fails to build on MacOs

### Background

The package `cargo-nextest` fails to build with our nix flake.  See this thread for more context: https://github.com/NixOS/nixpkgs/issues/456113

We should use the nixpkgs version which contains a patch, instead of building it from source.

Example error output:
```
$ nix develop -c cargo nextest run -p mpc-contract abi_has_not_changed               
error: Cannot build '/nix/store/6z9lm1yfyarw8h8fgfnllbcfk7bsxmq9-cargo-nextest-0.9.126.drv'.
       Reason: builder failed with exit code 101.
       Output paths:
         /nix/store/15fbg7arxzbg6qm7i17wnimzv62fd6ha-cargo-nextest-0.9.126
       Last 25 log lines:
       >     | |__________- in this macro invocation
       >     |
       >     = note: this error originates in the macro `crate::fire_usdt` (in Nightly builds, run with -Z macro-backtrace for more info)
       >
       > error[E0433]: failed to resolve: could not find `usdt_probes` in `usdt`
       >    --> /nix/var/nix/builds/nix-62407-102723561/cargo-nextest-0.9.126-vendor/nextest-runner-0.107.0/src/usdt.rs:222:23
       >     |
       > 222 |           $crate::usdt::usdt_probes::run__start!(|| {
       >     |                         ^^^^^^^^^^^ could not find `usdt_probes` in `usdt`
       >     |
       >    ::: /nix/var/nix/builds/nix-62407-102723561/cargo-nextest-0.9.126-vendor/nextest-runner-0.107.0/src/runner/dispatcher.rs:384:9
       >     |
       > 384 | /         crate::fire_usdt!(UsdtRunStart {
       > 385 | |             run_id: self.run_id,
       > 386 | |             profile_name: self.profile_name.clone(),
       > 387 | |             total_tests: test_list.test_count(),
       > ...   |
       > 392 | |             stress_duration_nanos,
       > 393 | |         });
       >     | |__________- in this macro invocation
       >     |
       >     = note: this error originates in the macro `crate::fire_usdt` (in Nightly builds, run with -Z macro-backtrace for more info)
       >
       > For more information about this error, try `rustc --explain E0433`.
       > error: could not compile `nextest-runner` (lib) due to 11 previous errors
       For full logs, run:
         nix-store -l /nix/store/6z9lm1yfyarw8h8fgfnllbcfk7bsxmq9-cargo-nextest-0.9.126.drv
error: Cannot build '/nix/store/p1shqvp1l9p852ady3jdjas9llvldhl3-nix-shell-env.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/mcgba3rysvsn6lh1zqq9ziwiswziwdqm-nix-shell-env
error: Build failed due to failed dependency
```

### User Story

_No response_

### Acceptance Criteria

make `cargo-nextest` work on MacOs with nix flake

### Resources & Additional Notes

_No response_

--- a/nix/cargo-tools.nix
+++ b/nix/cargo-tools.nix
@@ -3,7 +3,6 @@
   rustPlatform,
   fetchCrate,
   pkgs,
-  stdenv,
 }:
 
 let

